frontend/src/features/collection/Collection.tsx

```diff
@@ -9,47 +9,79 @@ import PageWithFooter from '../app/PageWithFooter'
 import WorkflowsTable from './WorkflowsTable'
 import Button from '../../chrome/Button'
 import Spinner from '../../chrome/Spinner'
+import SearchBox from '../search/SearchBox'
+import { selectSessionToken } from '../session/state/sessionState'
 
 const Collection: React.FC<any> = () => {
   const dispatch = useDispatch()
   const collection = useSelector(selectCollection)
   const loading = useSelector(selectIsCollectionLoading)
+  const token = useSelector(selectSessionToken)
+
+  const [ searchString, setSearchString ] = useState('')
+
+  const handleSearchChange = (q) => {
+    setSearchString(q)
+  }
 
   useEffect(() => {
-    loadCollection(dispatch, 1, 50)
+    dispatch(loadCollection(token))
   }, [dispatch])
 
   const [tableContainer, { height: tableContainerHeight }] = useDimensions();
 
+  // simple filter, comparing dataset reference with search input string
+  const filteredCollection = collection.filter((d) => {
+    return `${d.username}/${d.name}`.includes(searchString)
+  })
```

Comment:
```tsx
const collection = searchString ? filterVersionInfos(collection, searchString) : collection
```
define `filterVersionInfos` in `/src/qri/VersionInfo.ts` & import to this component

```ts
function filterVersionInfos(collection: VersionInfo[], searchString: string): VersionInfo[] {
  return collection.filter((d) => [
      d.username,
      d.name,
      d.metaTitle,
      d.themeList,
      d.path,
    ].findIndex((f: string) => f.includes(searchString)) > -1
  )
}
```